pkg/kubelet/userns_manager.go

```diff
@@ -0,0 +1,476 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"os"
+	"path"
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+const bitsUint32 = 32
```

Comment:
This is poorly named. If we're hard-coding a particular size of a `uid_t` then we should call this `hostUidBits` or something. If we want to know how many thirty two is, I don't think we should need a constant.

The right thing to do I think is to look up the right value at build time.

Reply:
If you have some uint type in Go (where you don't know the bit length) and you construct a zero value of it, the Golang `reflect` package lets you find out how many bits that is. Also, all unsigned ints can hold a zero.

Reply:
Ah, the reflect can't happen at compile time?

How about `linuxUidGidSizeInBits`?

Reply:
@sftim no, can't happen.

And again, this is not related to UID or GID size, at all. I don't know why you think that, so I don't know what else to say other than what I've just said. Can you please share why you think it is related to that? So I can either explain you where I think you are wrong, or understand what relationship I was not seeing that you are.

Reply:
This was renamed too